pkg/porter/credentials.go

```diff
@@ -106,22 +112,112 @@ func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
 	if err != nil {
 		return err
 	}
+
 	bundle, err := p.CNAB.LoadBundle(opts.CNABFile)
+	if err != nil {
+		return errors.Wrap(err, "failed to load bundle while generating new credentials")
+	}
+	_, err = p.generateNewCredentialSet(bundle, opts.Name, opts.Silent, opts.DryRun)
+	return err
+}
+
+func (p *Porter) chooseOrGenerateCredentialSet(bundle *bundle.Bundle) ([]string, error) {
 
+	credSets, err := p.Credentials.ReadAll()
 	if err != nil {
-		return err
+		return []string{}, errors.Wrap(err, "failed to read exisiting credential sets")
+	}
+
+	var credSetNames []string
+	for _, credSet := range credSets {
+		credSetNames = append(credSetNames, credSet.Name)
 	}
-	name := opts.Name
-	if name == "" {
-		name = bundle.Name
+
+	selectOptions := []string{generateCode, quitCode}
+	if len(credSetNames) > 0 {
+		selectOptions = append([]string{chooseCode}, selectOptions...)
+	}
+
+	var chooseOrGenerate string
+	selectChooseOrGeneratePrompt := &survey.Select{
+		Message: "Choose an option",
+		Options: selectOptions,
+		Default: selectOptions[0],
+	}
+
+	fmt.Fprintln(p.Out, "No credential set name passed")
+	survey.AskOne(selectChooseOrGeneratePrompt, &chooseOrGenerate, nil)
+
+	switch chooseOrGenerate {
+	case generateCode:
+		newCredSet, err := p.generateNewCredentialSet(bundle, "", false, false)
+		return []string{newCredSet}, err
+	case quitCode:
+		return []string{}, errors.New("Credentials are mandatory to install this bundle but none were provided with the `--cred` flag")
+	case chooseCode:
+		return p.chooseCredentialSet(bundle, credSetNames)
+	default:
+		return []string{}, errors.New("Unknow option")
+	}
+}
+
+func (p *Porter) chooseCredentialSet(bundle *bundle.Bundle, credSetNames []string) ([]string, error) {
+
+	var selectedCredSets []string
+
+	selectCredPrompt := &survey.MultiSelect{
+		Message: "Choose a set of credentials to use while installing this bundle",
+		Options: credSetNames,
+		Default: []string{bundle.Name},
+	}
+
+	err := survey.AskOne(selectCredPrompt, &selectedCredSets, nil)
+	if err != nil {
+		return []string{}, errors.New("no credential set selected")
 	}
+
+	if len(selectedCredSets) == 0 {
+		return []string{}, errors.New("no credential set selected")
+	}
+
+	return selectedCredSets, nil
+
+}
+
+func (p *Porter) generateNewCredentialSet(bundle *bundle.Bundle, credIdentifierName string, silent bool, dryRun bool) (string, error) {
+
+	if credIdentifierName == "" {
+		if silent {
+			credIdentifierName = bundle.Name
+		} else {
+			inputCredNamePrompt := &survey.Input{
+				Message: "Enter credential identifier name",
+				Default: bundle.Name,
+			}
+			survey.AskOne(inputCredNamePrompt, &credIdentifierName, nil)
+		}
+	}
+
 	genOpts := credentialsgenerator.GenerateOptions{
-		Name:        name,
+		Name:        credIdentifierName,
 		Credentials: bundle.Credentials,
-		Silent:      opts.Silent,
+		Silent:      silent,
+		DryRun:      dryRun,
 	}
+
+	// two extra spaces to align with survey prompts
```

Comment:
Is this comment still applicable/needed?

Reply:
Somehow i missed it.